src/webui/api/appcontroller.cpp

```diff
@@ -146,6 +146,11 @@ void AppController::preferencesAction()
     data["mail_notification_auth_enabled"] = pref->getMailNotificationSMTPAuth();
     data["mail_notification_username"] = pref->getMailNotificationSMTPUsername();
     data["mail_notification_password"] = pref->getMailNotificationSMTPPassword();
+
+    // Run an external program after torrent is added
+    data["start_autorun_enabled"] = pref->isStartAutoRunEnabled();
+    data["start_autorun_program"] = Utils::Fs::toNativePath(pref->getStartAutoRunProgram());
```

Comment:
There is another half (related to WebUI), care to implement it?

Reply:
Sure, my bad.

Reply:
I meant this: src/webui/www/private/views/preferences.html